src/shortestpaths/yen.jl

```diff
@@ -0,0 +1,102 @@
+"""
+    struct YenState{T, U}
+Designed for yen k-shortest-paths calculations.
+"""
+struct YenState{T, U<:Integer}<: AbstractPathState
+    dists::Vector{T}
+    paths::Vector{Vector{U}}
+end
+
+
+"""
+  yen_k_shortest_paths(g, source, target, distmx=DefaultDistance(), K);
```

Comment:
This should be indented 4 spaces, and the default value for `K` should be in the signature.